### DISCUSSION TITLE
Discussion on how to position nlmod among similar modeling aproaches

### DISCUSSION THREAD

**Discussion by bdestombe** [gwmod/nlmod#235] [General]
This alternative of the USGS is new to me. Maybe it's time to discuss the similar modeling aproaches to cherry pick from their routines. Or maybe suitable for us with minor contributions.
 - USGS (https://github.com/usgs/modflow-setup)
   - Early stages.
   - Maintained by the USGS
   - Uses a configuration file to configure a groundwater flow model
   - Targeted at US data sources
 - NHI
   - Structured grid 
 - Python IMOD (Deltares)

**Comment by dbrakenhoff** [comment]
My thoughts:

I think the USGS modflow_setup idea is pretty cool because it attempts to succinctly summarize which source data is used and in which way to build a modflow model. On the other hand, in each model I have worked on there is still always quite a bit of customization going on, which you want to write in a script. I'm sure there's support for that in their workflow, but I'm partial to just writing good script(s), instead of mixing a config file with custom scripts. Also adding a workflow manager (i.e. snakemake) to our modeling would be useful to further improve reproducibility and structuring of scripts. But we should definitely keep following their developments and see if anything would be useful to us. 

The imod package has some really nice stuff in it, and is well made, but it is an xarray-powered flopy replacement. Our philosophy is to follow flopy, which is inextricably linked with all Modflow 6 developments. Having said that, imod does contain some great functionality and since we're both using xarray, it would be great if our data format allows moving from one package to the other with minimal effort. Once we're a little further along with our development, we should put some effort into figuring out what steps are necessary to make interoperability possible. My hope is we both follow similar xarray-grid definitions so this step might just be formality. And I have used imod to draw cross-sections through DataArrays already, so I have an example lying around somewhere.

As for the NHI, we're going to have to see what that yields. From what I've seen it is likely part of the tools will work with IDF files. The imod python package provides methods for writing IDFs from xarray DataArrays. So once again I think it would be smart to put some focus on interoperability and help users to go back and forth between data formats that allow you to use specific tools. What that data format looks like is uncertain but definitely something we should look at once we know more.
